[16.0][MIG] account_invoice_import_simple_pdf

alexis-via opened the pull request.

alexis-via committed:
Initial check-in of account_invoice_import_simple_pdf

This new module is an improved alternative to the module account_invoice_import_invoice2data. Refer to README for more information.

alexis-via committed:
Temporary workaround of odoo bug https://github.com/odoo/odoo/issues/75492

Make pre-commit green

alexis-via committed:
Improve handling of start/end string cut when there are multiplue occurences of cut string

Add tests on string cut method

alexis-via committed:
[FIX] when the invoice has the VAT number of the supplier and also the VAT number of the customer

alexis-via committed:
[FIX] Read of the "Page Analysis" parameter

Before this fix, Page Analysis was always "First page" in the code.
FIX on view inheritance to avoid access right issues

alexis-via committed:
Improve date extraction when month is a string with accents

Support combining accent letters
Support bad accent extraction of mupdf on invoices such as Free mobile

alexis-via committed:
account_invoice_import_simple_pdf: coma -> comma (typo fix)

Migration scripts provided

alexis-via committed:
account_invoice_import_simple_pdf: easier extensibility for new fields

Use getattr to find the method which parses a particular field, as
suggested by Graeme Gellatly

oca-travis committed:
[UPD] Update account_invoice_import_simple_pdf.pot

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[FIX] account_invoice_import_simple_pdf: Remove exclude with invoice2data

alexis-via committed:
[FIX] account_invoice_import_simple_pdf: extract_rule position_min/position_max

Add list of fields on search_read() to improve perf

alexis-via committed:
account_invoice_import_simple_pdf: add onchange on date_format on field extract rules

alexis-via committed:
account_invoice_import_simple_pdf: install pymupdf by Debian package instead of pip

To goal: make travis green!

alexis-via committed:
same player try again

alexis-via committed:
Same player try again again

alexis-via committed:
Same player try again again again

alexis-via committed:
account_invoice_import_simple_pdf: support multiple tools for text extraction

alexis-via committed:
[FIX] access to form view of partners for users who are not accountants

Commit:
[UPD] Update account_invoice_import_simple_pdf.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_invoice_import_simple_pdf 14.0.2.0.0

etobella committed:
[FIX] account_invoice_import_simple_pdf: Fix view replace

flotho committed:
Use fix version of dateparser

FIX #569

OCA-git-bot committed:
account_invoice_import_simple_pdf 14.0.2.1.0

alexis-via committed:
account_invoice_import_simple_pdf: add apostrophe as thousand separator

Add test for apostrophe as thousand separator

alexis-via committed:
account_invoice_import_simple_pdf: parse July 5th, 2022 as date

Add tests such as "July 14th, 2022" and "1er Janvier 2022"

Commit:
[UPD] Update account_invoice_import_simple_pdf.pot

OCA-git-bot committed:
account_invoice_import_simple_pdf 14.0.2.2.0

alexis-via committed:
[FIX] simple_pdf: bad string

alexis-via committed:
account_invoice_import: improve handling of simple PDF invoices

Implement what is described in issue #610:
- account_invoice_import_simple_pdf and account_invoice_import_invoice2data can now be use
d together
- when importing an invoice for an unknown partner, it creates an empty invoice with PDF a
s attachment
- when importing an invoice via mail gateway and the from email is set
on a partner that doesn't have any import config, create a supplier
invoice with PDF as attachment and with that partner
- when importing an invoice via mail gateway and the from email is set
on a partner that has a single-line import config, create a supplier
invoice with PDF as attachment, that partner and 1 invoice line with price_unit=0.

Commit:
[UPD] Update account_invoice_import_simple_pdf.pot

OCA-git-bot committed:
account_invoice_import_simple_pdf 14.0.2.2.1

klodr committed:
Added translation using Weblate (French)

klodr committed:
Translated using Weblate (French)

Currently translated at 100.0% (144 of 144 strings)

Translation: edi-14.0/edi-14.0-account_invoice_import_simple_pdf
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import_simple_pdf/fr/

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (144 of 144 strings)

Translation: edi-14.0/edi-14.0-account_invoice_import_simple_pdf
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import_simple_pdf/fr/

alexis-via committed:
simple_pdf: add warning about constraint on regex version imposed by dateparser

alexis-via committed:
simple_pdf: use another invoice as test invoice

I don't use the Bouygues Telecom test invoice in the test suite, because
Bouygues Telecom also has an invoice2data template and invoice2data and
simple_pdf are now compatible so both modules are installed during the
test.

alexis-via committed:
simple_pdf: raise error if thousand sep = decimal sep

Commit:
[UPD] Update account_invoice_import_simple_pdf.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_invoice_import_simple_pdf 14.0.3.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-account_invoice_import_simple_pdf
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import_simple_pdf/

alexis-via committed:
[MIG] account_invoice_import_simple_pdf to v15

alexis-via committed:
simple_pdf: allow to match partners on additionnal fields

Add a hook to easily inherit the module
account_invoice_import_simple_pdf to match partners on fields other than
"vat". For example, the module l10n_fr_account_invoice_import_simple_pdf
takes advantage of this hook to match partners on SIREN (French
identification number for companies).

alexis-via committed:
[MIG] account_invoice_import_simple_pdf v15 -> v16

alexis-via committed:
simple_pdf: works with newer PyMuPDF versions

Add pypdf text extraction backend
Update README for pypdf, and also for the fact that PyMuPDF is now easy to install
Remove embedded XML file from PDF test file tests/pdf/akretion_france-test.pdf !
Add version of text extraction tools in test result

alexis-via committed:
account_invoice_import_simpler_pdf: remove pdfplumber support

pdfplumber depends on Pillow >= 9.1, whereas Odoo v14 depends on
Pillow==8.1.1. This causes python dependency issues. So, to make it
easier to deploy the module account_invoice_import_simple_pdf, I decided
to remove the support for pdfplumber.
Since october 2023, we have support for pypdf, which is also
a pure-python lib and doesn't depend on additional python libs, so it's
kind of a replacement for pdfplumber.

Replace sudo pip3 by pip3 in readme
Stop forcing version of dateparser, because it is now compatible with
latest version of regex. Update README accordingly.

icf20 commented:
is this abandoned?

alexis-via committed:
[FIX] account_invoice_import_simple_pdf: access to res.partner form view for users who are not in the invoicing group

alexis-via commented:
@icf20 no ! It works well. It depend on account_invoice_import that is not merged because I want to make important changes in it

icf20 commented:
> @icf20 no ! It works well.

thx for the reply, while my reply is late here it is, have to drop the interest in this the addon cause is lagging too much behind, odoo 18 (17.4) has electronic invoices and automatic tax reporting, so upgrade is a must before 2024-12-31, and this add-on is stuck on 14 or 16PR hell, i dont know if is intentional from odoo part but feels like they are pushing out all 3rd party companies offering services with their yearly upgrades
best of luck

alexis-via committed:
[FIX] simple_pdf: version extraction for fitz/PyMuPDF

Use with statement for NamedTemporaryFile()
add seek(0) to avoid error "empty file"
Add try/except on pypdf text extraction
Remove outdated comment in partner form view

victor-champonnois commented:
@alexis-via Hi, do you think this PR is usable in production, if we also deploy this https://github.com/OCA/edi/pull/753 released at a given stable commit ?

alexis-via committed:
[UDP] account_invoice_import_simple_pdf: adapt to big changes in account_invoice_import

alexis-via committed:
[IMP] account_invoice_import_simple_pdf: Use Command.create() in tests

alexis-via committed:
[IMP] account_invoice_import_simple_pdf: replace fitz by pymupdf in import

alexis-via committed:
[IMP] account_invoice_import_simple_pdf: improve partner form view

alexis-via committed:
[IMP] account_invoice_import_simple_pdf: make tests work when intrastat_base is installed

bguillot commented:
> @alexis-via Hi, do you think this PR is usable in production, if we also deploy this #753 released at a given stable commit ?

Hello, The answer is late but we use it in production at Akretion. hope it helps

bguillot reviewed (approved):
Used in production, works correctly

alexis-via commented:
/ocabot merge nobump
